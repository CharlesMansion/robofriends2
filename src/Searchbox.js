import React from 'react';


const Searchbox = () => {
	return (
		<div className='pa2'>
		<input type='search' placeholder='Search robots...'></input>
		</div>
		);
};
export default Searchbox;