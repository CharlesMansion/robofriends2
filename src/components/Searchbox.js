import React from 'react';


const Searchbox = ({searchChange}) => {
	return (
		<div className='pa2'>
			<label for="search">
			<input 
			className='pa3 ba b--blue bg-light-yellow' 
			type='search' 
			placeholder='Search robots...'
			onChange={searchChange}
			id="search">
			</input>
			</label>
		</div>
		);
};
export default Searchbox;