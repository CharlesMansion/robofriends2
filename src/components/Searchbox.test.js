import React from 'react';
import Searchbox from './Searchbox';
import {shallow} from 'enzyme';

const mockSearchange = jest.fn();

const wrapper = shallow(<Searchbox searchange={mockSearchange}/>);

it('expects Searchbox to render correctly', () => {
	expect(wrapper).toMatchSnapshot();
});


