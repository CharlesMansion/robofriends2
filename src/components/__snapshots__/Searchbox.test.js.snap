// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expects Searchbox to render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Searchbox
    searchange={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <label
        for="search"
      >
        <input
          aria-label="Search"
          className="pa3 ba b--blue bg-light-yellow"
          id="search"
          placeholder="Search robots..."
          type="search"
        />
      </label>,
      "className": "pa2",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <input
          aria-label="Search"
          className="pa3 ba b--blue bg-light-yellow"
          id="search"
          placeholder="Search robots..."
          type="search"
        />,
        "for": "search",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "aria-label": "Search",
          "className": "pa3 ba b--blue bg-light-yellow",
          "id": "search",
          "onChange": undefined,
          "placeholder": "Search robots...",
          "type": "search",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      "type": "label",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <label
          for="search"
        >
          <input
            aria-label="Search"
            className="pa3 ba b--blue bg-light-yellow"
            id="search"
            placeholder="Search robots..."
            type="search"
          />
        </label>,
        "className": "pa2",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <input
            aria-label="Search"
            className="pa3 ba b--blue bg-light-yellow"
            id="search"
            placeholder="Search robots..."
            type="search"
          />,
          "for": "search",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "aria-label": "Search",
            "className": "pa3 ba b--blue bg-light-yellow",
            "id": "search",
            "onChange": undefined,
            "placeholder": "Search robots...",
            "type": "search",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "type": "label",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
